Mathlib/FieldTheory/SeparableClosure.lean

```diff
@@ -0,0 +1,386 @@
+  ⟨fun h ↦ (isSeparable_def _ _).2 fun _ ↦ (mem_separableClosure_iff F E).1 (h ▸ mem_top),
+    fun h ↦ top_unique fun x _ ↦ (mem_separableClosure_iff F E).2 (((isSeparable_def _ _).1 h) x)⟩
+
+/-- If `K / E / F` is a field extension tower, then `separableClosure F K` is contained in
+`separableClosure E K`. -/
+theorem separableClosure.le_restrictScalars [Algebra E K] [IsScalarTower F E K] :
+    separableClosure F K ≤ (separableClosure E K).restrictScalars F := fun x hx ↦ by
+  simp only [mem_restrictScalars, mem_separableClosure_iff] at hx ⊢
+  exact hx.map_minpoly E
+
+/-- If `K / E / F` is a field extension tower, such that `E / F` is separable, then
+`separableClosure F K` is equal to `separableClosure E K`. -/
+theorem separableClosure.eq_restrictScalars_of_isSeparable [Algebra E K] [IsScalarTower F E K]
+    [IsSeparable F E] : separableClosure F K = (separableClosure E K).restrictScalars F := by
+  refine le_antisymm (separableClosure.le_restrictScalars F E K) fun x hx ↦ ?_
+  rw [mem_restrictScalars, mem_separableClosure_iff,
+    ← isSeparable_adjoin_simple_iff_separable] at hx
+  haveI : IsSeparable F (E⟮x⟯.restrictScalars F) := IsSeparable.trans F E E⟮x⟯
+  have h : x ∈ E⟮x⟯.restrictScalars F := mem_adjoin_simple_self E x
+  exact (mem_separableClosure_iff F _).2 <| separable_of_mem_isSeparable F _ h
+
+/-- If `K / E / F` is a field extension tower, then `E` adjoin `separableClosure F K` is contained
+in `separableClosure E K`. -/
+theorem separableClosure.adjoin_le [Algebra E K] [IsScalarTower F E K] :
+    adjoin E (separableClosure F K) ≤ separableClosure E K :=
+  adjoin_le_iff.2 <| le_restrictScalars F E K
+
+/-- A compositum of two separable extensions is separable. -/
+instance IntermediateField.isSeparable_sup (L1 L2 : IntermediateField F E)
+    [h1 : IsSeparable F L1] [h2 : IsSeparable F L2] :
+    IsSeparable F (L1 ⊔ L2 : IntermediateField F E) := by
+  rw [← le_separableClosure_iff] at h1 h2 ⊢
+  exact sup_le h1 h2
+
+/-- A compositum of separable extensions is separable. -/
+instance IntermediateField.isSeparable_iSup {ι : Type*} {t : ι → IntermediateField F E}
+    [h : ∀ i, IsSeparable F (t i)] : IsSeparable F (⨆ i, t i : IntermediateField F E) := by
+  simp_rw [← le_separableClosure_iff] at h ⊢
+  exact iSup_le h
+
+end separableClosure
+
+namespace Field
+
+/-- The (infinite) separable degree for a general field extension `E / F` is defined
+to be the degree of `(separableClosure F E) / F`. -/
+def sepDegree := Module.rank F (separableClosure F E)
+
+/-- The (infinite) inseparable degree for a general field extension `E / F` is defined
+to be the degree of `E / (separableClosure F E)`. -/
+def insepDegree := Module.rank (separableClosure F E) E
+
+/-- The (finite) inseparable degree for a general field extension `E / F` is defined
+to be the degree of `E / (separableClosure F E)` as a natural number. It is defined to be zero
+if such field extension is infinite. -/
```

Comment:
You write `separableClosure F E` without the parentheses in the [module docstring](https://github.com/leanprover-community/mathlib4/pull/9338/files#diff-fbae6aa6a45b5781c17dc0397d1edfbcf1ed608887c0c6aa87cf2f0a57d44145R24-R34) but use parentheses here. (I think they're not needed.)